File: 3869-smallest-index-with-digit-sum-equal-to-index/3869-smallest-index-with-digit-sum-equal-to-index.c
int smallestIndex(int* nums, int numsSize) {
    for (int i = 0; i < numsSize; i++) {
        int sum = 0;
        int num = nums[i];
        while (num) {
            sum += num % 10;
            if (sum > i)
                break;
            num /= 10;
        }
        if (sum == i)
            return i;
    }
    return -1;
}

File: 3869-smallest-index-with-digit-sum-equal-to-index/3869-smallest-index-with-digit-sum-equal-to-index.go
func smallestIndex(nums []int) int {
    n := len(nums)
    for i := 0; i < n; i++ {
        num, sum := nums[i], 0
        for num > 0 {
            sum += num % 10
            if sum > i{
                break
            }
            num /= 10
        }
        if (sum == i){
            return i
        }
    }
    return -1
}